wp-content/themes/core/components/section_nav/Section_Nav_Controller.php

```diff
@@ -0,0 +1,220 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Project\Templates\Components\section_nav;
+
+use Tribe\Libs\Utils\Markup_Utils;
+use Tribe\Project\Templates\Components\Abstract_Controller;
+use Tribe\Project\Templates\Components\button\Button_Controller;
+use Tribe\Project\Templates\Components\Deferred_Component;
+use Tribe\Project\Templates\Components\navigation\Navigation_Controller;
+use Tribe\Project\Templates\Components\text\Text_Controller;
+
+class Section_Nav_Controller extends Abstract_Controller {
+
+	public const ATTRS             = 'attrs';
+	public const CLASSES           = 'classes';
+	public const CONTAINER_ATTRS   = 'container_attrs';
+	public const CONTAINER_CLASSES = 'container_classes';
+	public const MENU              = 'menu';
+	public const MOBILE_LABEL      = 'mobile_label';
+	public const DESKTOP_LABEL     = 'desktop_label';
+	public const MORE_LABEL        = 'more_label';
+	public const STICKY            = 'sticky';
+	public const MOBILE_INIT_OPEN  = 'mobile_init_open';
+
+	/**
+	 * @var string[]
+	 */
+	private array $attrs;
+
+	/**
+	 * @var string[]
+	 */
+	private array $classes;
+
+	/**
+	 * @var string[]
+	 */
+	private array $container_attrs;
+
+	/**
+	 * @var string[]
+	 */
+	private array $container_classes;
+
+	/**
+	 * @var int|string|\WP_Term
+	 */
+	private $menu;
+
+	/**
+	 * @var string
+	 */
+	private string $mobile_label;
+
+	/**
+	 * @var string
+	 */
+	private string $desktop_label;
+
+	/**
+	 * @var string
+	 */
+	private string $more_label;
+
+	/**
+	 * @var bool
+	 */
+	private bool $sticky;
+
+	/**
+	 * @var bool
+	 */
+	private bool $mobile_init_open;
+
+	/**
+	 * @var string
+	 */
+	private string $unique_id;
+
+	/**
+	 * @var string
+	 */
+	private string $container_id;
+
+	/**
+	 * @var string
+	 */
+	private string $more_id;
+
+	public function __construct( array $args = [] ) {
+		$this->unique_id    = uniqid();
```

Comment:
❓ 
Curious what we need a UUID for?

Reply:
Good question. For the ARIA attributes required to associate a button with the thing it controls, so the mobile show/hide button and the container that is shown/hidden; and [the "More" button](https://github.com/moderntribe/square-one/pull/888/files#diff-830a1fc2c6e6373e01e4eb1df77af46ae9a258e201b2819d5d144dfe8df4cc01R152) for showing/hiding whatever menu items are tucked into [the drawer container on desktop](https://github.com/moderntribe/square-one/pull/888/files#diff-830a1fc2c6e6373e01e4eb1df77af46ae9a258e201b2819d5d144dfe8df4cc01R208).

Since the ARIA association works with the `id` attribute, we need a unique "id" for each section nav on a page.

Reply:
Makes sense, it  does feel a-little inefficient to generate a completely new UUID for every execution though. Unless there is another way to do an ID key on the component, when there will possibly be multiple of them on a page, let's leave it as-is.

Reply:
[ACF provides a unique identifier](https://www.advancedcustomfields.com/resources/blocks/) for each block, but it looks like we aren't passing it out of Base_Model. I think we should add `Id` to the `name`, `classes`, `data`, and `mode` there, and then we won't need this step and could just call `$this->id` for anything that requires it.

Reply:
Thanks @ChrisMKindred , while it's true, the ACF block ID will work for instances which are defined via a block, the component could just as easily be used as a direct template part in a single page or post template where we wouldn't have access to a block ID. I suppose we could modify the component to accept a unique ID of some sort and then pass along the block ID or require it when defining the usage outside that context. But, it seems just as easy to allow the each component to set their own UUID at the time of usage.  We use [this same `uniqid()` approach pretty regularly throughout SquareOne](https://github.com/moderntribe/square-one/search?q=uniqid%28) & the projects we've built previously.